File: python/fucking_coffee.py
#!/usr/bin/env python

import sys
import subprocess
import telnetlib
import time

# exit if no sessions with my username are found
output = subprocess.check_output('who')
if 'my_username' not in output:
    sys.exit()

time.sleep(17)

coffee_machine_ip = '10.10.42.42'
password = '1234'
password_prompt = 'Password: '

con = telnetlib.Telnet(coffee_machine_ip)
con.read_until(password_prompt)
con.write(password + "\n")

# Make some coffee!
con.write("sys brew\n")
time.sleep(24)

# love the smell!
con.write("sys pour\n")
con.close()    

File: go/fucking-coffee.go
package main

import (
	"fmt"
	"log"
	"os"
	"regexp"
	"time"

	"github.com/codeskyblue/go-sh"
	"github.com/google/goexpect"
)

func main() {
	// exit early if no sessions with my username are found
	currentUser, _ := sh.Command("who").Command("grep", "my_username").Output()
	if currentUser == nil {
		os.Exit(1)
	}

	// info about the coffee machine
	coffeeMachineIP := "10.10.42.42"
	password := "1234"
	passwordPrompt := "Password: "
	delayBeforeBrew := 17 * time.Second
	delay := 24 * time.Second

	// timeout for the telnet prompts
	timeout := 10 * time.Minute

	// sleep 17 seconds before brewing coffee
	time.Sleep(delayBeforeBrew)

	// spawn a new telnet session with the coffee machine
	t, _, err := expect.Spawn(fmt.Sprintf("telnet %s", coffeeMachineIP), -1)
	if err != nil {
		log.Fatal(err)
	}
	defer t.Close()

	t.Expect(regexp.MustCompile(passwordPrompt), timeout)
	t.Send(password + "\n")
	t.Expect(regexp.MustCompile("telnet>"), timeout)
	t.Send("sys brew\n")
	time.Sleep(delay)
	t.Expect(regexp.MustCompile("telnet>"), timeout)
	t.Send("sys pour\n")
	t.Expect(regexp.MustCompile("telnet>"), timeout)
	t.Send("exit\n")
}
